### PR TITLE
Fix performance factor breaking stamina recovery

### PR DESCRIPTION
**Pull request by BaerMitUmlaut**
**When merged this pull request will:**
- Fix performance factor breaking stamina recovery

From the original VBS documentation:

---

**Acidosis accumulation**
`anaerobicFatigue += anaerobicDwork * maxPowerFatigueRatio * 1.1;`
[...]
`maxPowerFatigueRatio` is a constant designed to scale the accumulation and recovery of anaerobicFatigue: `maxPowerFatigueRatio = 0.057 / peakPower;`

---

They might mean that it's a constant regardless of the actual peakPower, which is calculated from the VO₂max (which in turn is adjusted by the performance factor). This seems to fix issue #6163.

### CODE REVIEW

**Comment by BaerMitUmlaut** [comment]
I don't think it's how it's meant either, this is more of a hotfix tbh. If you have an idea where else the problem might be let me know.

**Comment by ulteq** [comment]
I think part of the problem is that we derive `_perceivedFatigue` off `anReserve`. Which is not how it is meant to be done.

Instead we should use `anFatique` directly: https://github.com/acemod/ACE3/pull/5723/files#diff-0bdf0f3e6417972d88363532f5354a5aR114

I'll try to do some tests tomorrow.

**Comment by BaerMitUmlaut** [comment]
That's definitely not the underlying issue. The stamina bar displays only the anaerobic reserve which very visibly shows that it does essentially not recover at all. `_perceivedFatigue` is only used for feedback effects and has no influence on the recovery.

**Comment by ulteq** [comment]
> That's definitely not the underlying issue.

Right.

New attempt:

What if the performance factor would also modify `ANTPERCENT`.

File: addons/advanced_fatigue/functions/fnc_mainLoop.sqf [hunk 1]
```diff
@@ -53,7 +53,9 @@ GVAR(ae1Reserve) = GVAR(ae1Reserve) - _ae1Power / WATTSPERATP;
 GVAR(ae2Reserve) = GVAR(ae2Reserve) - _ae2Power / WATTSPERATP;
 GVAR(anReserve)  = GVAR(anReserve)  -  _anPower / WATTSPERATP;
 // Increase anearobic fatigue
-GVAR(anFatigue)  = GVAR(anFatigue) + _anPower * (0.057 / GVAR(peakPower)) * 1.1;
+// 1264.64 = 0.057 / peakPower with default VO2 max
+// See #6163
+GVAR(anFatigue)  = GVAR(anFatigue) + _anPower * 1264.64 * 1.1;
```
Comment (ulteq) [inline addons/advanced_fatigue/functions/fnc_mainLoop.sqf]: `1264.64` seems to be very high. Where did you get peakPower `0.00004507211` from?

**Comment by PabstMirror** [comment]
Just to clarify my changes `GVAR(peakPower) = 1264` under default values.

But I'm not sure if this is correct, line 58 increases fatigue and line 69 handles recovery.
Before both lines used the term `(0.057 / GVAR(peakPower)` to scale; this goes down as performance goes up.
This means that you had more reserves, but it also took longer to recover.

With this PR we change fatigue increase to ignore peakPower, but recovery still is negatively effected by it.

**Comment by ulteq** [comment]
I'm still not convinced by this PR. But I sadly don't have time to take a closer look.

One solution might be to finish this PR: https://github.com/acemod/ACE3/pull/5723

It should be almost ready to go if we leave the [Experimental movement speed limiter](https://github.com/acemod/ACE3/pull/5723/commits/18661ebe02014d6f2c91350e341cf01eeddc8f0e) out.

**Comment by Mike-MF** [comment]
Closing due to age and/or inactivity, unlikely to be finished or has to be reworked due to other changes over the years. May be re-opened if someone finds it useful, even better just open a new up-to-date PR.

**Pull request closed without merging**